packages/ibm-products/src/components/ConditionBuilder/ConditionBlock/ConditionBlock.js

```diff
@@ -0,0 +1,247 @@
+import React, { useContext, useState } from 'react';
+import { Close } from '@carbon/react/icons';
+import { ConditionBuilderItem } from '../ConditionBuilderItem/ConditionBuilderItem';
+import { pkg } from '../../../settings';
+import PropTypes from 'prop-types';
+
+import cx from 'classnames';
+import ConditionConnector from '../ConditionConnector/ConditionConnector';
+
+import { ConditionBuilderContext } from '../ConditionBuilderContext/DataTreeContext';
+import { ConditionBuilderButton } from '../ConditionBuilderButton/ConditionBuilderButton';
+
+/**
+ * This component build each block of condition consisting of property, operator value and close button.
+ */
+const blockClass = `${pkg.prefix}--condition-builder`;
+
+function ConditionBlock(props) {
+  let { property, value, operator } = props.state;
+  let {
+    onRemove,
+    state,
+    conjunction,
+    onConnectorOperatorChange,
+    isStatement,
+    group,
+    aria,
+  } = props;
+  const { inputConfig } = useContext(ConditionBuilderContext);
+
+  const [showDeletionPreview, setShowDeletionPreview] = useState(false);
+
+  //filtering the current property to access its properties and config options
+  const { icon, type, label } =
+    inputConfig.properties?.filter(
+      (eachProperty) =>
+        eachProperty.label?.toUpperCase() == property?.toUpperCase()
+    )[0] ?? {};
+
+  return (
+    <div
+      className={cx(
+        `${blockClass}__condition-block conditionBlockWrapper ${blockClass}__gyap`
+      )}
+      role="row"
+      tabIndex={-1}
+      aria-level={aria.level}
+      aria-posinset={aria.posinset}
+      aria-setsize={aria.setsize}
+    >
+      {/* this code draws the connector conditions between each condition block */}
+      {conjunction && (
+        <ConditionConnector
+          className={`${blockClass}__gyap ${blockClass}__gyap-bottom`}
+          role="gridcell"
+          operator={conjunction}
+          onChange={(op) => onConnectorOperatorChange(op)}
+        />
+      )}
+
+    {/* this code draws the starting statement for each group */}
+      {isStatement && (
+        <div
+          className={`${blockClass}__gyap-bottom ${blockClass}__gyap`}
+          role="gridcell"
+        >
+          <ConditionBuilderItem
+            open={false}
+            label={group.statement}
+            title={'Condition'}
```

Comment:
continue to be wary of hardcoded strings

Reply:
used translations at missed out places